File: 1458/1458.c
int maxDotProductOfTwoSubsequences(struct TreeNode* root) {
    if (!root) return 0;
    int l = maxDotProductOfTwoSubsequences(root->left);
    int r = maxDotProductOfTwoSubsequences(root->right);
    return 1 + (l > r ? l : r);
}

File: 1458/1458.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDotProductOfTwoSubsequences = function(root) {
    if (!root) return 0;
    return 1 + Math.max(maxDotProductOfTwoSubsequences(root.left), maxDotProductOfTwoSubsequences(root.right));
};